src/test_typing_extensions.py

```diff
@@ -4402,6 +4402,39 @@ class Cat(Animal):
             'voice': str,
         }
 
+    @skipIf(sys.version_info == (3, 14, 0, "beta", 1), "Broken on beta 1, fixed in beta 2")
+    def test_inheritance_pep563(self):
+        def _make_td(future, class_name, annos, base, extra_names=None):
+            lines = []
+            if future:
+                lines.append('from __future__ import annotations')
+            lines.append('from typing import TypedDict')
+            lines.append(f'class {class_name}({base}):')
+            for name, anno in annos.items():
+                lines.append(f'    {name}: {anno}')
+            code = '\n'.join(lines)
+            ns = {**extra_names} if extra_names else {}
+            exec(code, ns)
+            return ns[class_name]
+
```

Comment:
This is all very magical. Wouldn't it make more sense to just hard-code these four fairly short code samples?

Reply:
Ah, it's eight. So maybe use a template at least?

Reply:
I feel that would be more complicated; it would require conditionals on half the lines of the template and `textwrap.dedent`.